meeteval/wer/wer/time_constrained.py

```diff
@@ -386,27 +438,46 @@ def time_constrained_siso_word_error_rate(
         reference_pseudo_word_level_timing: strategy for pseudo-word level timing for reference
         hypothesis_pseudo_word_level_timing: strategy for pseudo-word level timing for hypothesis
         collar: collar applied to hypothesis pseudo-word level timings
+        reference_sort: How to sort the reference. Options: 'segment', 'word', True, False. See below
+        hypothesis_sort: How to sort the reference. Options: 'segment', 'word', True, False. See below
+        
+    Reference / Hypothesis sorting options:
+    - True: sort by segment start time and assert that the word-level timings are sorted by start time
+    - False: do not sort and don't check word order
+    - 'segment': sort by segment start time and don't check word order
+    - 'word': sort by word start time
+
+    >>> time_constrained_siso_word_error_rate(TimeMarkedTranscript(['a b', 'c d'], [(0,2), (0,2)]), TimeMarkedTranscript(['a'], [(0,1)]))
+    ErrorRate(error_rate=0.75, errors=3, length=4, insertions=0, deletions=3, substitutions=0, reference_self_overlap=SelfOverlap(overlap_time=4, total_time=4, overlap_rate=1.0), hypothesis_self_overlap=SelfOverlap(overlap_time=0, total_time=1, overlap_rate=0.0))
     """
     reference = TimeMarkedTranscript.create(reference)
     hypothesis = TimeMarkedTranscript.create(hypothesis)
 
-    reference = get_pseudo_word_level_timings(reference, reference_pseudo_word_level_timing)
-    hypothesis = get_pseudo_word_level_timings(hypothesis, hypothesis_pseudo_word_level_timing, collar)
+    reference_, _ = sort_and_validate(reference, reference_sort, reference_pseudo_word_level_timing, 'reference')
+    hypothesis_, _ = sort_and_validate(hypothesis, hypothesis_sort, hypothesis_pseudo_word_level_timing, 'hypothesis')
+
+    hypothesis = apply_collar(hypothesis, collar)
 
-    return _time_constrained_siso_error_rate(
-        reference.transcript, hypothesis.transcript,
-        reference.timings, hypothesis.timings,
+    er = _time_constrained_siso_error_rate(
+        reference_.transcript, hypothesis_.transcript,
+        reference_.timings, hypothesis_.timings,
     )
+    er = replace(
```

Comment:
Add a comment like: `pseudo_word_level_timing and collar change the time stamps, hence calculate the overlap with the original time stamps`

Reply:
Done